File: backend/src/main/java/com/pairingplanet/pairing_planet/domain/enums/CookingTimeRange.java
package com.pairingplanet.pairing_planet.domain.enums;

/**
 * Cooking time ranges for recipes.
 * Uses approximate ranges since cooking time varies by skill level.
 */
public enum CookingTimeRange {
    UNDER_15_MIN,    // Quick snacks, salads
    MIN_15_TO_30,    // Fast weeknight meals
    MIN_30_TO_60,    // Standard home cooking (default)
    HOUR_1_TO_2,     // Elaborate dishes
    OVER_2_HOURS     // Slow roasts, fermented dishes
}

File: frontend_mobile/lib/core/constants/cooking_time_range.dart
/// Cooking time range options for recipes.
/// Uses approximate ranges since cooking time varies by skill level.
enum CookingTimeRange {
  under15min('UNDER_15_MIN'),
  min15to30('MIN_15_TO_30'),
  min30to60('MIN_30_TO_60'),
  hour1to2('HOUR_1_TO_2'),
  over2hours('OVER_2_HOURS');

  final String code;
  const CookingTimeRange(this.code);

  /// Get the translation key for this cooking time range
  String get translationKey => 'recipe.cookingTime.$code';

  /// Get CookingTimeRange from backend code string
  static CookingTimeRange fromCode(String? code) {
    if (code == null) return CookingTimeRange.min30to60;
    return CookingTimeRange.values.firstWhere(
      (e) => e.code == code,
      orElse: () => CookingTimeRange.min30to60,
    );
  }
}
